### DISCUSSION TITLE
Linux:  build&run osu with dotnet 6.0.0

### DISCUSSION THREAD

**Discussion by biando** [ppy/osu#16149] [Q&A]
git clone https://github.com/ppy/osu.git
cd osu
dotnet publish -c Release -r linux-x64 --self-contained osu.Desktop

The Folder osu/osu.Desktop/bin/Release/net5.0/linux-x64/ now contains "osu" to play

**Comment by bdach** [comment]
what is the purpose of this discussion? are you trying to report that the game does not build with .NET 6?

if so, we know. before migrating to .NET 6 we have to eliminate the dependency on entity framework. see: https://github.com/ppy/osu/discussions/15544

**Reply by biando** [reply]
???

**Reply by bdach** [reply]
i had the exact same reaction to your opening post. can you explain why did you open this discussion?

**Reply by biando** [reply]
Howto "Linux: build&run osu with dotnet 6.0.0 "

**Reply by bdach** [reply]
i'm still lost. if you wish to play the game there are binary releases [on the release page](https://github.com/ppy/osu/releases). you will not be able to play multiplayer with self-built releases.

**Reply by biando** [reply]
there is a possibility for "build & run" with dotnet 6.0, which currently does not work with "dotnet run --project osu.Desktop" (https://github.com/ppy/osu)

**Comment by smoogipoo** [comment]
The proper way is to install .NET 5 SDK. Readme will be updated in https://github.com/ppy/osu/pull/16163

**Reply by biando** [reply]
"dotnet publish -c Release -r linux-x64 --self-contained osu.Desktop" includes NET5 functionality like https://github.com/ppy/osu/releases/latest/download/osu.AppImage
